src/main/java/it/pn/frontend/e2e/pages/mittente/ApiKeyPAPage.java

```diff
@@ -427,36 +436,38 @@ public void siVisualizzaPopUpVisualizza() {
         try {
             By subTitleVisualizzaBy = By.xpath("//p[contains(@data-testid,'subtitle-top')]");
             getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(subTitleVisualizzaBy));
-            logger.info("Si visualizza correttamente il pop up visualizza");
+            logger.info("Si visualizza correttamente il sotto titolo del pop up");
         }catch (TimeoutException e){
-            logger.error("NON si visualizza correttamente il pop up visualizza");
-            Assert.fail("NON si visualizza correttamente il pop up visualizza");
+            logger.error("NON si visualizza correttamente il sotto titolo pop up: "+e.getMessage());
+            Assert.fail("NON si visualizza correttamente il sotto titolo pop up: "+e.getMessage());
         }
     }
 
     public void chiudiPopUpVisualizza() {
-        this.getWebDriverWait(30).until(ExpectedConditions.elementToBeClickable(this.closeButtonPopUpVisualizza));
+        this.getWebDriverWait(30).withMessage("il Bottone chiudere pop up non è cliccabile")
+                .until(ExpectedConditions.elementToBeClickable(this.closeButtonPopUpVisualizza));
         this.closeButtonPopUpVisualizza.click();
     }
 
     public void clickVisualizzaIdApiKey() {
-        this.getWebDriverWait(30).until(ExpectedConditions.elementToBeClickable(this.visualizzaIdGruppo));
+        this.getWebDriverWait(30).withMessage("Il bottone visualizza Id api key non è cliccabile")
+                .until(ExpectedConditions.elementToBeClickable(this.visualizzaIdGruppo));
         this.visualizzaIdGruppo.click();
     }
 
     public void popUpGruppiAssociati() {
         try{
             By popUpGruppiAssociatiBy = By.xpath("//h2[contains(text(),'Gruppi associati alla API')]");
             getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(popUpGruppiAssociatiBy));
```

Comment:
inserire messaggio

Reply:
il messaggio è gestito con try and catch

Reply:
non vedo try catch però

Reply:
guarda linea 464   :  <img width="1125" alt="Screenshot 2023-11-21 at 11 43 36" src="https://github.com/pagopa/pn-frontend-e2e/assets/138457809/bc17fdee-0c3d-4bfa-ab72-aa3401851b78">